test-apps/imodel-transformer/src/ElementUtils.ts

```diff
@@ -0,0 +1,74 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import { DbResult, Id64Set, Id64String } from "@bentley/bentleyjs-core";
+import {
+  Category, CategorySelector, DisplayStyle, ECSqlStatement, GeometricModel3d, IModelDb, ModelSelector, SubCategory,
+} from "@bentley/imodeljs-backend";
+
+export namespace ElementUtils {
+
+  function queryElementIds(iModelDb: IModelDb, classFullName: string): Id64Set {
+    const elementIds = new Set<Id64String>();
+    iModelDb.withPreparedStatement(`SELECT ECInstanceId FROM ${classFullName}`, (statement: ECSqlStatement) => {
+      while (DbResult.BE_SQLITE_ROW === statement.step()) {
+        elementIds.add(statement.getValue(0).getId());
+      }
+    });
+    return elementIds;
+  }
+
+  export function validateModelSelectors(iModelDb: IModelDb): void {

```

Comment:
I am personally a fan of abstracting the `validateSomethings` functions like so, but this is debatable. If you don't reasonably expect any more element classes to be added to these validation utils, I'd ignore this.

```tsx
export function makeValidateElementType<T extends Element>(
  elementClass: EntityClassType<Element>,
  doValidate: (elem: Element) => void,
): void {
  return function validator (iModelDb: IModelDb) {
    const ids = queryElementIds(iModelDb, elementClass.classFullName);
    ids.forEach((id) => {
      const elem= iModelDb.elements.getElement<Element>(id, elementClass);
      doValidate(elem);
    });
  };
}

const validateModelSelectors = makeValidateElementType<ModelSelector>(ModelSelector, validateModelSelector);
const validateCategorySelectors = makeValidateElementType<CategorySelector>(CategorySelector, validateCategorySelector);
```

Reply:
Elements that put most of their important content in JsonProperties and those that have Element Ids (that need to be remapped) embedded in that JSON are tricky to clone. The validation is really about making sure that the source Element was structured properly prior to transformation and transformation didn't mess things up or carry over invalid Ids. These types of problems are hard to detect. I added a validation subset based on an issue that I was investigating. Easier to catch problems with validation than to diagnose them after the fact.

I suggest that we add additional validation as needed over time. Once we have a larger set of methods, a better pattern will probably emerge. For now, I am going to leave them alone since they have served their immediate purpose of diagnosing a problem and confirming a fix.